src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+    }
+
+    /**
+     * Retrieve device system information, including hardware info and lan status
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getSystemInfo(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(SYSTEM), JsonNode.class);
+        if (response == null) {
+            return;
+        }
+        addStatisticsProperty(statistics, "System#Serial Number", response.get("serialNumber"));
+        addStatisticsProperty(statistics, "System#Software Version", response.get("softwareVersion"));
+        addStatisticsProperty(statistics, "System#System State", response.get("state"));
+        addStatisticsProperty(statistics, "System#System Name", response.get("systemName"));
+        addStatisticsProperty(statistics, "System#System Uptime", response.get("uptime"));
+        addStatisticsProperty(statistics, "System#System Build", response.get("build"));
+        addStatisticsProperty(statistics, "System#System Reboot Needed", response.get("rebootNeeded"));
+        addStatisticsProperty(statistics, "System#Device Model", response.get("model"));
+        addStatisticsProperty(statistics, "System#Device Hardware Version", response.get("hardwareVersion"));
+
+        if (response.get("lanStatus") == null) {
+            return;
+        }
+        addStatisticsProperty(statistics, "Lan Status#Duplex", response.get("lanStatus").get("duplex"));
+        addStatisticsProperty(statistics, "Lan Status#Speed Mbps", response.get("lanStatus").get("speedMbps"));
+        addStatisticsProperty(statistics, "Lan Status#State", response.get("lanStatus").get("state"));
+    }
+
+    /**
+     * Retrieve list of device's applications
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getApplications(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(APPS), JsonNode.class);
+        if (response.get("apps") == null) {
+            return;
+        }
+        response.get("apps").forEach(jsonNode -> {
+            statistics.put("Applications#" + getJsonProperty(jsonNode, "appName").asText() + " Version",
+                    getJsonProperty(jsonNode, "versionInfo").asText());
+            statistics.put("Applications#" + getJsonProperty(jsonNode, "appName").asText() + " Last Updated",
+                    new Date(getJsonProperty(jsonNode, "lastUpdatedOn").asLong()).toString());
+        });
+    }
+
+    /**
+     * Retrieve list of active sessions on the device
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getSessions(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(SESSIONS), JsonNode.class);
+        if (response.get("sessionList") == null) {

```

Comment:
"response.get("sessionList")" can be extracted and reused

Reply:
Please review class and try to extract such calls to local variables, and reuse them...I will not comment on any future ones